subsys/bootloader/bl_validation/bl_validation.c

```diff
@@ -22,29 +22,60 @@ int set_monotonic_version(uint16_t version, uint16_t slot)
 	printk("Setting monotonic counter (version: %d, slot: %d)\r\n",
 		version, slot);
 
-	int err = set_monotonic_counter((version << 1) | !slot);
+	uint16_t num_cnt_slots;
+	int err;
+
+	err = num_monotonic_counter_slots(BL_MONOTONIC_COUNTERS_DESC_NSIB, &num_cnt_slots);
+	if (err != 0) {
+		printk("Failed during reading of the number of the counter slots, error: %d\r\n",
+		       err);
+		return err;
+	}
 
-	if (num_monotonic_counter_slots() == 0) {
+	if (num_cnt_slots == 0) {
 		printk("Monotonic version counter is disabled.\r\n");
-	} else if (err != 0) {
-		printk("set_monotonic_counter() error: %d\n\r", err);
+		return -EINVAL;
+	}
+
+	err = set_monotonic_counter(BL_MONOTONIC_COUNTERS_DESC_NSIB, (version << 1) | !slot);
+	if (err != 0) {
+		printk("Failed during setting the monotonic counter, error: %d\r\n", err);
 	}
+
 	return err;
 }
 
-uint16_t get_monotonic_version(uint16_t *slot_out)
+int get_monotonic_version(uint16_t *version_out)
 {
-	uint16_t monotonic_version_and_slot = get_monotonic_counter();
+	uint16_t monotonic_version_and_slot;
+	int err;
 
-	bool slot = !(monotonic_version_and_slot & 1);
-	uint16_t version = monotonic_version_and_slot >> 1;
+	if (version_out == NULL) {
+		return -EINVAL;
+	}
 
-	if (slot_out) {
-		*slot_out = slot;
+	err = get_monotonic_counter(BL_MONOTONIC_COUNTERS_DESC_NSIB, &monotonic_version_and_slot);
+	if (err) {
+		return err;
 	}
-	return version;
+
+	*version_out = monotonic_version_and_slot >> 1;
+
+	return err;
 }
 
+int get_monotonic_slot(uint16_t *slot_out)
+{
+	uint16_t monotonic_version_and_slot = get_monotonic_counter();
```

Comment:
`get_monotonic_counter()` is missing arguments.

Reply:
fixed